Lysko denis

DenLysko opened the pull request:
Прошу принять мой PR

DenLysko committed:
feat: Изначальное задание

DenLysko committed:
feat: моё решение

DenLysko committed:
docs, fix: переименование метода

alexandraroots commented on omd.py:
```diff
@@ -0,0 +1,30 @@
+        return step2_umbrella()
+    return step2_no_umbrella()
+
+if __name__ == '__main__':
```
– 2 Нарушение pep8 (2 строки перед main), но лучше пользоваться линетрами (black/isort/flake8, остальные по любви и желанию)

alexandraroots commented on omd.py:
```diff
@@ -0,0 +1,30 @@
+
+def step2_no_umbrella():
+    print(
+        f"По пути в бар была солнечная погода 🌞! \n"
```
– 2 Разных типы кавычек
И f строки не стоит использовать там, где в них нет необходимости. Тут просто строки, в которые не вставляются какие-либо переменные, и литер black будет как раз на такое ругаться

alexandraroots reviewed (commented):
5/10

DenLysko committed:
style: used black

DenLysko committed:
feat: добавление csv-файла для задания

DenLysko committed:
feat: Основные методы, связанные с меню

DenLysko committed:
feat: Методы связанные с обработкой логики вариантов

DenLysko committed:
feat: Вынес методы в отдельный файл

Эти методы объединяет то, что они работают с csv, и поэтому, например, import csv можно убрать из menu.py

DenLysko committed:
refactor: import'ы

DenLysko committed:
refactor: Переписал if-else на match и вынес

DenLysko committed:
refactor: изменил название метода и вынес reader из метода

DenLysko committed:
refactor: Добавление докстрингов и аннотаций

DenLysko committed:
feat: Преобразование отчёт для csv-файла

Содержимое отчёта не поменялось. Просто более правильное разбитие по колонкам csv-файла

alexandraroots commented on Python_Lecture04_Homework/menu.py:
```diff
@@ -0,0 +1,53 @@
+                # Не было указаний до какого числа необходимо огруглять среднюю зарплату,
+                # поэтому вывожу число полностью. Но всегда можно написать round(value[4], 2), например
+                print(
+                    f'Департамент "{key}". Численность: {value[0]}, минимальная зарплата: {value[2]}, '
```
Разнородные кавычки

alexandraroots commented on Python_Lecture04_Homework/methods_for_options.py:
```diff
@@ -0,0 +1,93 @@
+import csv
```
Импортирована библиотека, но не использована

alexandraroots commented on Python_Lecture04_Homework/menu.py:
```diff
@@ -0,0 +1,53 @@
+    my_reader = writer_and_reader.get_reader()
+    option = get_option()
+
+    match option:
```
Получается пользователь может только 1 раз выбрать опцию. Чаще всего меню предполагает что-то типо: я как пользователь сначала выбираю посмотреть на иерархию, потом на статистику, а потом еще и сохранить все. Без перезапуска программы
Сюда бы хорошо лег while True

alexandraroots commented on Python_Lecture04_Homework/menu.py:
```diff
@@ -0,0 +1,53 @@
+
+    option = 0
+    options = [1, 2, 3]
+    while option not in options:
```
Если я тут наберу случайно любой символ != 1 2 3, то не очень понятно, а что происходит то. Лучше сообщить пользователю что-то типо: вы ввели неверный номер из меню

alexandraroots commented on Python_Lecture04_Homework/menu.py:
```diff
@@ -0,0 +1,53 @@
+            report = methods_for_options.generate_report(my_reader)
+            for key, value in report.items():
+                # Не было указаний до какого числа необходимо огруглять среднюю зарплату,
+                # поэтому вывожу число полностью. Но всегда можно написать round(value[4], 2), например
```
Можно, но это не будет нести бизнес смысла. Так что лучше округлить

alexandraroots commented on Python_Lecture04_Homework/writer_and_reader.py:
```diff
@@ -0,0 +1,69 @@
+    :param value: Значение из словаря
+    :return: Строка для csv-файла
+    """
+    another_row: list[str] = []
```
Кажется, что можно воспользоваться генератором. Ниже аналогично

alexandraroots commented on Python_Lecture04_Homework/menu.py:
```diff
@@ -0,0 +1,53 @@
+
+    match option:
+        case 1:
+            departments_and_commands = methods_for_options.generate_command_hierarchy(
```
правильное будет импортировать конкретные методы, а не весь модуль
```python
from methods_for_options import generate_command_hierarchy
``` 

И дальше уже обращаться к конкретному методу без указания его модуля. Во-первых так читаемые, во-вторых не нужно будет тянуть лишние методы, которые в этом файле могут и не использоваться

alexandraroots reviewed (commented):
Хорошее разделение на модули

-2 за разнородные кавычки и неиспользованный импорт (стоит пользоваться линтерами)
-2 за отсутсвие подсказки, что пользователь ввел не верный пункт из меню и завершение работы после каждого выбора пункта (неинтуитивно поведение)

6/10

DenLysko committed:
docs: Task

DenLysko committed:
feat: HomeWork #3

alexandraroots commented on Python_Lecture08_Homework/CountVectorize.py:
Лучше, конечно, создавать отдельный пр под каждое дз, не очень удобно проверять

alexandraroots replied:
Хорошее решение!
9/10 (-1 б за просроченный дд)
